cli/src/cmd/forge/script/build.rs

```diff
@@ -0,0 +1,189 @@
+use crate::{cmd::get_cached_entry_by_name, compile, opts::forge::ContractInfo};
+
+use ethers::{
+    prelude::{
+        artifacts::Libraries, cache::SolFilesCache, ArtifactId, Project, ProjectCompileOutput,
+    },
+    solc::artifacts::{CompactContractBytecode, ContractBytecode, ContractBytecodeSome},
+    types::{Address, U256},
+};
+
+use foundry_utils::PostLinkInput;
+use std::{collections::BTreeMap, str::FromStr};
+
+use super::*;
+
+impl ScriptArgs {
+    /// Compiles the file with auto-detection and compiler params.
+    pub fn build(&mut self, script_config: &ScriptConfig) -> eyre::Result<BuildOutput> {
+        let (project, output) = self.get_project_and_output(script_config)?;
+
+        let (contracts, sources) = output.into_artifacts_with_sources();
```

Comment:
needs to be updated with latest ethers-rs changes to get the `SourceFile` from the artifact directly